crates/ruma-client-api/src/discovery/get_supported_versions.rs

```diff
@@ -59,37 +59,35 @@ impl Response {
     /// The versions returned will be sorted from oldest to latest. Use [`.find()`][Iterator::find]
     /// or [`.rfind()`][DoubleEndedIterator::rfind] to look for a minimum or maximum version to use
     /// given some constraint.
-    pub fn known_versions(&self) -> impl DoubleEndedIterator<Item = MatrixVersion> {
+    pub fn known_versions(&self) -> BTreeSet<MatrixVersion> {
```

Comment:
Should have returned `impl IntoIterator<IntoIter: DoubleEndedIterator>` originally, then this wouldn't have been a breaking change 😄 

(just kidding - [a] associated type bounds weren't a thing back then [b] it's ugly [c] it would have been awkward to use)

Reply:
If we want to avoid the breaking change we could still do `.into_iter()`. But since there are other breaking changes, I thought it wouldn't matter much.